jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -383,6 +400,7 @@ ecma_check_value_type_is_spec_defined (ecma_value_t value) /**< ecma value */
                 || ecma_is_value_boolean (value)
                 || ecma_is_value_number (value)
                 || ecma_is_value_string (value)
+                || ECMA_ASSERT_VALUE_IS_SYMBOl (value)
```

Comment:
Due to the macro guards IMO we cant.

Since `ecma_is_value_symbol` is only available the the the Symbol builtin is enabled, the proper guard for the assert would be:

```c
JERRY_ASSERT (ecma_is_value_undefined (value)
                || ecma_is_value_null (value)
                || ecma_is_value_boolean (value)
                || ecma_is_value_number (value)
                || ecma_is_value_string (value)
#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
                || ecma_is_value_symbol(value)
#endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */
                || ecma_is_value_object (value));
```

but is result the following error:

```
home/fancsikrobi/harmony/jerryscript/jerry-core/ecma/base/ecma-helpers-value.c:403:1: error: embedding a directive within macro arguments is not portable [-Werror]
 #ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
 ^
/home/fancsikrobi/harmony/jerryscript/jerry-core/ecma/base/ecma-helpers-value.c:405:1: error: embedding a directive within macro arguments is not portable [-Werror]
 #endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */
 ^
```

Since JERRY_ASSSERT is also a macro.

Reply:
Does it make sense that make `ecma_is_value_symbol` available whether `CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN` defined or nor?
If `CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN` is set `ecma_is_value_symbol` could return false directly.

Reply:
I am ok with the macro.